aten/src/ATen/core/op_registration/op_registration.cpp

```diff
@@ -165,4 +165,54 @@ RegisterOperators::~RegisterOperators() = default;
 RegisterOperators::RegisterOperators(RegisterOperators&&) noexcept = default;
 RegisterOperators& RegisterOperators::operator=(RegisterOperators&&) noexcept = default;
 
+
+CppFunction::CppFunction(KernelFunction func, std::unique_ptr<c10::FunctionSchema> schema)
+  : func_(std::move(func))
+  , schema_(std::move(schema))
+  {}
+
+Module::Module(const char* ns)
+  : ns_(ns)
+  {}
+
+Module::Module(Module&&) noexcept = default;
+Module& Module::operator=(Module&&) noexcept = default;
+
+
+Module&& Module::def(const char* schema) && {
+  register_.op(c10::RegisterOperators::options()
+    .schema(schema)
+    .aliasAnalysis(c10::AliasAnalysisKind::FROM_SCHEMA));
+  return std::move(*this);
+}
+
+static std::string addNamespace(const char* ns, const char* unqual_name_or_schema) {
+  if (ns) {
+    // TODO: slow!  Fix internal data structures so I don't have to paste the
+    // names together
+    std::ostringstream oss;
+    oss << ns << "::" << unqual_name_or_schema;
+    return oss.str();
+  } else {
+    return unqual_name_or_schema;
+  }
+}
+
+Module&& Module::def(const char* unqual_name, CppFunction&& f) && {
+  register_.op(c10::RegisterOperators::options()
```

Comment:
I think calling .def() multiple times for the same operator should be a hard error. The current implementation allows it and merges.

Reply:
I agree, but this increases the scope of the PR substantially, as I now have to start modifying how the existing RegisterOperators mechanism works to enforce this behavior. My preference is for this issue to resolve itself "automatically" when schema merging gets implemented, in a separate PR. If you like, we can block this PR until schema merging lands, but in this case, the impermissible behavior is very mild and I think it's OK to land this first, and schema merging later. What do you think?

Reply:
You could hack it together by just remembering a global (or static within Module) std::vector of all ops that got registered with `.def()`. That should be relatively small scope and not require changing the RegisterOperators mechanism.

I think we should do that before landing this to avoid people starting to use the impermissible behavior and running into bc issues once we remove it.

Reply:
But this is all wasted code when schema merging gets written. Backwards compatibility is not a do or die thing; you can (and should!) declare APIs to be unstable while you're still working on them (which is the case here!), otherwise you are just making a lot of busy work for yourself, without any benefit!

Reply:
One thing I can do is file an issue describing all of the "over-permissiveness" we need to audit and test for before moving the API out of beta.